tests/main/user-libnss/task.yaml

```diff
@@ -0,0 +1,50 @@
+summary: Ensure our osutil.Find{Uid,Gid} code work with libnss
+
+description: |
+    The os/user code in go will behave differently when compiled with
+    or without cgo. This is confusing so we created the helpers
+    osutil.Find{Uid,Gid} that automatically fall back to calling
+    getent(1) when build without cgo. This test ensures these are
+    working correctly.
+
+# only run on a well defined system where we know how to setup libnss
+systems: [ubuntu-18.04-64]
+
+prepare: |
+    echo "Save nsswitch.conf"
+    cp /etc/nsswitch.conf /etc/nsswitch.conf.save
+    echo "Install extrausers"
+    apt install libnss-extrausers
+    echo "Enable libnss-extrusers"
+    sed -i 's/^group:.*compat/\0 extrausers/' /etc/nsswitch.conf
+    sed -i 's/^passwd:.*compat/\0 extrausers/' /etc/nsswitch.conf
+    sed -i 's/^shadow:.*compat/\0 extrausers/' /etc/nsswitch.conf
+    echo "Workaround silly bug that causes extrausers to crash when missing"
+    for name in gshadow shadow; do
+        touch /var/lib/extrausers/$name
+        chmod 640 /var/lib/extrausers/$name
+        chown root:shadow /var/lib/extrausers/$name
+    done
+    echo "Add user"
+    adduser --extrausers --disabled-login --no-create-home --gecos '' --uid 9876 --shell /bin/false extratest
+
+restore: |
+    mv /etc/nsswitch.conf.save /etc/nsswitch.conf
+    apt autoremove -y libnss-extrausers
```

Comment:
Should run `userdel --extrausers extratest` here

Reply:
might be easier to just rm -rf /var/lib/extrausers